src/slurm_plugin/clustermgtd.py

```diff
@@ -766,6 +834,58 @@ def _is_node_healthy(self, node, private_ip_to_instance_map):
                 )
             ) and self._is_node_state_healthy(node, private_ip_to_instance_map)
 
+    def _is_node_bootstrap_failure(self, node, private_ip_to_instance_map):
+        """
+        Check if a slurm node has boostrap failure.
+
+        Here's the cases of bootstrap error we are checking:
+        Bootstrap error that causes instance to self terminate, info about these nodes.
+        Bootstrap error that prevents instance from joining cluster but does not cause self termination.
+        """
+        if node.is_static:
+            # Node is currently in replacement and no backing instance
+            if (
+                not ClusterManager._is_backing_instance_valid(
+                    node, instance_ips_in_cluster=list(private_ip_to_instance_map.keys())
+                )
+                and node.name in self._static_nodes_in_replacement
+            ):
+                log.warning("Node bootstrap error: Node is currently in replacement and no backing instance")
```

Comment:
Shall we add an information of what node you are referring to for all the log messages in this function?